### PR TITLE
chore: Merges master into dev

### PR DESCRIPTION
**Pull request by EspenAlbert**
## Description

Merges master into dev

Link to any related issue(s): CLOUDP-262464

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

**Review by lantoli — APPROVED** [review]
nice!

**Comment by lantoli** [comment]
TestAccBackupCompliancePolicy_overwriteBackupPolicies fail seems a real issue

**Comment by marcosuma** [comment]
@EspenAlbert did you have to resolve any conflicts? Is [this](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2443/commits/1198aea1665e031d4e5c986aeb530fe2f1e04fb0) the only fix you had to do?

**Comment by EspenAlbert** [comment]
@marcosuma trying to describe the changes below

# Merging Strategy

Using `{master_name}/{dev_name}` below:
E.g., `checkSingleProvider/checkReplicaSetAWSProvider`
- function named `checkSingleProvider` in master
- function named `checkReplicaSetAWSProvider` in dev

<br>

- SDK Imports
  - `go.mongodb.org/atlas-sdk/v20240530002/admin` vs `admin20231115 "go.mongodb.org/atlas-sdk/v20231115014/admin"` Choosing to always keep the `dev` version
  - `admin.XXX/admin20231115.XXX` choosing to always keep the `dev` version
- Using `acc.GetClusterInfo`, keeping the master version
  - `cloud_backup_schedule_test.go`
  - `cluster_outage_simulation_test.go`
  - +++
- specific files:
  - `advanced_cluster_test.go`
    - `checkSingleProvider/checkReplicaSetAWSProvider` new checks in both master and dev
    - `checkMultiCloudSharded/checkSingleShardedMultiCloud`
      - `external_id` check in dev
      - a lot of `disk_iops` checks in master
      - keeping combination
  - `acc.cluster.go`
    - upgrade to use the new admin without replication spec

**Comment by EspenAlbert** [comment]
### Test failures

tl;dr: All the consistent failures have **tracking tickets** (except for the 1st failure `TestMigAdvancedCluster_v0StateUpgrade_ReplicationSpecs`)

Here are all the tests that have "never" passed in the branch:

#### advanced_cluster
- `TestMigAdvancedCluster_v0StateUpgrade_ReplicationSpecs` has 4 failures:
  - @AgustinBettati do you have context on this?
  - Failing with: `[{1 Argument is deprecated This parameter is deprecated. Please refer to our examples, documentation, and 1.18.0 migration guide for more details at https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/1.18.0-upgrade-guide.html.markdown [{{} disk_size_gb}]}]`
  - [an hour ago failed in 1 second](https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/9993986747/job/27622955663#step:5:145)
  - [an hour ago failed in 1 second](https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/9993531302/job/27621347248#step:5:145)
  - [3 hours ago failed in 1 second](https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/9992318182/job/27617264037#step:5:145)
  - [5 hours ago failed in 1 second](https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/9990332907/job/27611015302#step:5:145)
- `TestAccClusterAdvancedCluster_singleShardedMultiCloud` has 4 failures:
  - [Ticket tracking the fix for invalid azure disk size](https://jira.mongodb.org/browse/CLOUDP-258714)
  - Failing with: `POST: HTTP 400 Bad Request (Error code: "DISK_SIZE_INVALID_FOR_AZURE") Detail: The disk size for a cluster with Azure nodes must correspond to an Azure disk size. The valid Azure disk sizes are 8, 16, 32, 64, 128, 256, 512, 1024, 2048, or 4095 GB. Reason: Bad Request. Params: []`
  - [an hour ago failed in 5 seconds](https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/9993986747/job/27622955663#step:5:227)
  - [an hour ago failed in 6 seconds](https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/9993531302/job/27621347248#step:5:228)
  - [3 hours ago failed in 6 seconds](https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/9992318182/job/27617264037#step:5:249)
  - [5 hours ago failed in 8 seconds](https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/9990332907/job/27611015302#step:5:250)

#### backup
- `TestAccBackupRSCloudBackupSchedule_copySettings` has 4 failures:
  - [Tracking ticket for "support" replication_spec.*.id again](https://jira.mongodb.org/browse/CLOUDP-262014)
  - Failing with: `Detail: Received JSON for the copySettings.java.util.ArrayList[0].replicationSpecId attribute does not match expected format. Reason: Bad Request. Params: [copySettings.java.util.ArrayList[0].replicationSpecId]`
  - [58 minutes ago failed in 13 minutes](https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/9993986747/job/27622957038#step:5:183)
  - [an hour ago failed in 12 minutes](https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/9993531302/job/27621348729#step:5:182)
  - [2 hours ago failed in 13 minutes](https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/9992318182/job/27617262791#step:5:184)
  - [5 hours ago failed in 13 minutes](https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/9990332907/job/27611014432#step:5:182)

#### generic
- `TestAccBackupCompliancePolicy_overwriteBackupPolicies` has 4 failures:
  - [Tracking ticket for "support" replication_spec.*.id again](https://jira.mongodb.org/browse/CLOUDP-262014)
  - Failing with: `Detail: Received JSON for the copySettings.java.util.ArrayList[0].replicationSpecId attribute does not match expected format. Reason: Bad Request. Params: [copySettings.java.util.ArrayList[0].replicationSpecId]`
  - [an hour ago failed in 14 minutes](https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/9993986747/job/27622959619#step:5:178)
  - [an hour ago failed in 15 minutes](https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/9993531302/job/27621351527#step:5:178)
  - [2 hours ago failed in 14 minutes](https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/9992318182/job/27617266978#step:5:178)
  - [5 hours ago failed in 12 minutes](https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/9990332907/job/27611017869#step:5:178)

**Review by jwilliams-mongo — APPROVED** [review]

**Review by lantoli — APPROVED** [review]
@EspenAlbert nice, thanks for the detailed info

**Review by AgustinBettati — APPROVED** [review]
LGTM, only remaining change is to include fix for `TestMigAdvancedCluster_v0StateUpgrade_ReplicationSpecs`. To provide more details this failing test was missed in the PR where we deprecated attributes in the schema (https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2421). One option is to adjust the test  to only fail on schema errors.
Great job with the PR and all details provided!

**Comment by EspenAlbert** [comment]
Merging this now. FYI @lantoli @AgustinBettati @maastha @marcosuma :

1. Cancelled the acceptance tests for the last commit (27511cc) to avoid extensive resource usage.
2. Added fix for `TestMigAdvancedCluster_v0StateUpgrade_ReplicationSpecs` in [27511cc](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2443/commits/27511cc0dcb315d69e9aa9aca218233bb1d606b2)
  - Local run: `PASS: TestMigAdvancedCluster_v0StateUpgrade_ReplicationSpecs (0.00s)`
 3. Skipped: `TestAccBackupRSCloudBackupSchedule_copySettings` and `TestAccBackupCompliancePolicy_overwriteBackupPolicies` in e78c924
 4. Last [full code health](https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/10004622242) looks promising only 
   - `TestAccClusterAdvancedCluster_singleShardedMultiCloud` fails (same as before)
   - `TestMigAdvancedCluster_v0StateUpgrade_ReplicationSpecs` fails (fixed in 27511cc (see 2))
